### ISSUE TITLE
"last kill" & "last loss" does not reflect reality

### ISSUE THREAD

**Issue by ezi0o** [WhiteRusssian/PySpy#10]
win7 - pyspy 0.4

the "last kill" & "last loss" information are wrong and always shows at least 10+ days compared to reality.

exemple: i killed 4 ships yesterday, yet pyspy shows 19d, and my last loss was 7 days ago and pyspy shows 19d too : [https://i.imgur.com/Ly69XcB.jpg](url)

i have checked with many other corp mates and pypsy display wrong data for everyone

**Comment by Sadario** [comment]
As far as I can see, this seems to be data that gets pulled from the proprietary DB. Tried finding a fix, but it seems to be in project owners hands, if I'm not mistaken.

**Comment by WhiteRusssian** [comment]
Hi guys, apologies, I have been travelling. @Sadario is right, this is not a problem with the PySpy client.

My personal server that downloads all killmails and then processes those killmails to check for cyno losses etc. also provides the "last kill" & "last loss" data points. The problem is, this analysis takes about 8 hours to run on my mac mini and I feel like running it everyday is overkill.

For your background, I am downloading all killmails into a local mongoDB (currently 43GB), from here I am creating summary killmails into a SQLite3 DB (21GB) on an SSD. On this dataset i perform analyses. The most CPU intensive job is to figure out the average number of attackers for all kills that a given pilot was involved in (takes about 6 out of the 8 hours). Again, this is a feature i rarely use myself but it was requested by other users and i enjoyed the challenge of implementing it. Once all analyses are done I have two summary data sets, one for each victim and one for each attacker (in total <200MB), which i upload to an online database and this database is what PySpy runs queries against.

I personally do not use last kill and last loss myself. I am more interested in the cyno loss functionality and here I was never too concerned about update intervals of >1 week. 

I am going to look into finding a way to maybe rely on zkill for last loss / kill data and will report back here in due course. Thanks for your patience.

**Comment by Sadario** [comment]
> The most CPU intensive job is to figure out the average number of attackers for all kills that a given pilot was involved in (takes about 6 out of the 8 hours).

Maybe change how that works? Make it calculate it for the last 6/12 months? '+=' all that shit, then divide by number of killmails.

**Comment by WhiteRusssian** [comment]
I have looked at my server logs again and realised that instead of just analysing those character_ids for which kill/loss events had occurred since the last update, my server was running a full analysis on all character_ids over all killmail_ids since the beginning of time. This was due to some simply coding error which corrected today. Going forward, the PySpy server will update daily at 0100 Sydney time. Note, as this was not a client issue, there is no need for an updated PySpy release.

I will leave this issue open for the next few days until we can all confirm that last loss / last kill is indeed working correctly now. Thanks for pointing this out to me.
